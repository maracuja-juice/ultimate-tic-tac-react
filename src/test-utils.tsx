// See: https://testing-library.com/docs/react-testing-library/setup#custom-render
import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { IconProvider } from './components/Icons';
import '@testing-library/jest-dom'; // https://github.com/testing-library/jest-dom/issues/282#issuecomment-660034778
import { MemoryRouter } from 'react-router-dom';

const AllTheProviders: FC = ({ children }) => {
  return (
    <IconProvider>
      <MemoryRouter>{children}</MemoryRouter>
    </IconProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
