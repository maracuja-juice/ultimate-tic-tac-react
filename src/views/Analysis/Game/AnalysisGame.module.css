.analysisLayout {
  font-size: 16px;
  margin-bottom: 1em;

  display: grid;
  grid-row-gap: 0.5em;
  grid-column-gap: 1.5em;
  grid-template-areas: 'moveList' 'historyButtons' 'game';
  grid-template-rows: minmax(50vmin, 70vmin) minmax(min-content, 7vmin) 0.3fr;
}

.moveList {
  grid-area: moveList;
  padding: 0;
  overflow-y: scroll;
  position: relative;
}

.analysisGame {
  grid-area: game;
  width: 85vmin;
  justify-self: center;
}

.historyButtons {
  grid-area: historyButtons;
  display: flex;
  justify-content: center;
}

@media screen and (min-width: 640px) and (max-height: 768px),
  screen and (min-width: 1025px) {
  .analysisLayout {
    margin: 0.25em 0 0.25em;

    grid-template-columns: minmax(min-content, 1fr) minmax(80vmin, 85vmin);
    grid-template-rows: min-content 85vmin;
    grid-template-areas: '. historyButtons' 'moveList game';
  }

  .applyHeight {
    height: 100%;
  }
}

@media screen and (min-width: 1000px) {
  .analysisLayout {
    padding: 1em 1.5em 0.5em 1.5em;
    margin-bottom: 0.5em;
  }

  .historyButtons {
    margin-bottom: 1em;
  }
}
