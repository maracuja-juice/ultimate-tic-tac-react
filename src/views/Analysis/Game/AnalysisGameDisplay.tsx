import * as React from 'react';
import { BigBoard } from '../../../components/Board/BigBoard/BigBoard';
import { AnalysisGame } from '../../../state/AppState';
import { HistoryButtons } from '../../../components/Analysis/HistoryButtons/HistoryButtons';
import { MoveList } from '../../../components/Analysis/MoveList/MoveList';
import { scroller } from 'react-scroll/modules';
import { moveScrollElementBaseName } from '../../../components/Analysis/ScrollElementConstants';
import styles from './AnalysisGame.module.css';

interface AnalysisGameDisplayProps {
  moveForwardInHistory: (numberOfMoves: number) => void;
  moveBackwardInHistory: (numberOfMoves: number) => void;
  analysisGame: AnalysisGame;
}

function scrollToElement(moveNumberToScrollTo: number) {
  scroller.scrollTo(moveScrollElementBaseName + moveNumberToScrollTo, {
    duration: 300,
    smooth: true,
    containerId: 'moveList',
    // Thanks to this, when scrolling to an element, some elements above it will also be displayed
    offset: -125,
  });
}

export function AnalysisGameDisplay(props: AnalysisGameDisplayProps) {
  const analysisGame = props.analysisGame;
  const reversedMoves = analysisGame.moves.slice().reverse();
  const currentlyAppliedMove =
    reversedMoves[
      reversedMoves.findIndex((m) => m.moveNumber === analysisGame.currentMove)!
    ];

  return (
    <div className={styles.analysisLayout}>
      <div id="moveList" className={styles.moveList}>
        <MoveList
          currentMove={analysisGame.currentMove}
          reversedMoves={reversedMoves}
          moveForwardInHistory={props.moveForwardInHistory}
          moveBackwardInHistory={props.moveBackwardInHistory}
        />
      </div>
      <div className={styles.historyButtons}>
        <HistoryButtons
          currentMove={analysisGame.currentMove}
          lastMove={reversedMoves[0] && reversedMoves[0].moveNumber}
          moveForwardInHistory={props.moveForwardInHistory}
          moveBackwardInHistory={props.moveBackwardInHistory}
          onInteraction={scrollToElement}
        />
      </div>
      <div className={styles.analysisGame}>
        <BigBoard
          board={analysisGame.board}
          activeBoards={analysisGame.activeBoards}
          currentPlayer={analysisGame.game.currentPlayer}
          markTileSpecially={{
            condition: currentlyAppliedMove !== undefined,
            position: currentlyAppliedMove
              ? {
                  boardPosition: currentlyAppliedMove.boardPosition,
                  tilePosition: currentlyAppliedMove.tilePosition,
                }
              : undefined,
          }}
        />
      </div>
    </div>
  );
}
