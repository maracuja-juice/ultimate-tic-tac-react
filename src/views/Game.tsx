import * as React from 'react';
import BigBoard from '../components/Board/BigBoard/BigBoard';
import GameFinishedDisplay from '../components/GameFinishedDisplay/GameFinishedDisplay';

interface GameProps {
}

export class Game extends React.Component<GameProps> {

    render() {
        return (
            <div className="center">
                <GameFinishedDisplay />
                <BigBoard/>
            </div>
        );
    }
}