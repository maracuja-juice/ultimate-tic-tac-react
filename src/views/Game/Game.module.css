.gameWrapper {
  display: grid;
  width: 50%;
  margin-top: 1em;
  margin-bottom: 1em;

  font-size: 7px;
}

@media only screen and (min-width: 500px) {
  .gameWrapper {
    font-size: 12px;
  }
}

@media only screen and (min-width: 700px) {
  .gameWrapper {
    font-size: 16px;
  }
}
