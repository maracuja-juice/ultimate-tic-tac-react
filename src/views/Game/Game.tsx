import * as React from 'react';
import { BigBoard } from '../../components/Board/BigBoard/BigBoard';
import GameFinishedDisplay from '../../components/GameFinishedDisplay/GameFinishedDisplay';
import { AppState, Player, SmallBoardInformation } from '../../state/AppState';
import { playerMoved } from '../../state/currentGame/game/gameAction';
import { connect } from 'react-redux';
import { Point } from '../../util';
import styles from './Game.module.css';
import classNames from 'classnames';

interface GameProps {
  currentPlayer: Player;
  board: SmallBoardInformation[];
  activeBoards: Point[];
  onPlayerMoved: (
    boardX: number,
    boardY: number,
    tileX: number,
    tileY: number,
  ) => void;
}

export function Game(props: GameProps) {
  const { currentPlayer, board, activeBoards, onPlayerMoved } = props;

  return (
    <div className={classNames('centerAll', styles.applyHeight)}>
      <div className={styles.gameWrapper}>
        <GameFinishedDisplay />
        <BigBoard
          currentPlayer={currentPlayer}
          board={board}
          activeBoards={activeBoards}
          onPlayerMoved={onPlayerMoved}
        />
      </div>
    </div>
  );
}

const mapStateToProps = (state: AppState) => ({
  currentPlayer: state.currentGame.game.currentPlayer,
  board: state.currentGame.board,
  activeBoards: state.currentGame.activeBoards,
});

const mapDispatchToProps = (dispatch: any) => ({
  onPlayerMoved: (
    boardX: number,
    boardY: number,
    tileX: number,
    tileY: number,
  ) => dispatch(playerMoved({ x: boardX, y: boardY }, { x: tileX, y: tileY })),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Game);
