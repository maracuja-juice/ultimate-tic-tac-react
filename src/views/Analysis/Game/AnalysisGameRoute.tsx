import * as React from 'react';
import { useEffect } from 'react';
import { RouteComponentProps } from 'react-router';
import { AnalysisGame, AppState } from '../../../state/AppState';
import { connect } from 'react-redux';
import {
  loadFinishedGameByDate,
  loadFinishedGameById,
  loadLatestFinishedGame,
  moveBackwardInHistory,
  moveForwardInHistory,
} from '../../../state/analysisGame/analysisGameActions';
import routes from '../../../routes/routes';
import { AnalysisGameDisplay } from './AnalysisGameDisplay';
import { NoGameFound } from '../NoGameFound';
import { parseJSON, isValid } from 'date-fns';

interface AnalysisGameRouteProps
  extends RouteComponentProps<{ param: string }> {
  analysisGame: AnalysisGame;
  loadAnalysisGameById: (id: string) => void;
  loadLatestAnalysisGame: () => void;
  loadAnalysisGameByDate: (date: Date) => void;
  moveForwardInHistory: (numberOfMoves: number) => void;
  moveBackwardInHistory: (numberOfMoves: number) => void;
}

export function AnalysisGameRoute(props: AnalysisGameRouteProps) {
  const pathName = props.location.pathname;
  const param = props.match.params.param;
  const {
    loadLatestAnalysisGame,
    loadAnalysisGameById,
    loadAnalysisGameByDate,
    analysisGame,
  } = props;

  useEffect(() => {
    if (pathName.includes(routes.AnalysisLatest)) {
      loadLatestAnalysisGame();
    } else if (isValid(parseJSON(param))) {
      loadAnalysisGameByDate(parseJSON(param));
    } else {
      loadAnalysisGameById(param);
    }
  }, [
    param,
    pathName,
    loadAnalysisGameById,
    loadAnalysisGameByDate,
    loadLatestAnalysisGame,
  ]);

  return analysisGame.board.length > 0 ? (
    <AnalysisGameDisplay
      moveForwardInHistory={props.moveForwardInHistory}
      moveBackwardInHistory={props.moveBackwardInHistory}
      analysisGame={props.analysisGame}
    />
  ) : (
    <NoGameFound tag="h1" center={true} />
  );
}

const mapStateToProps = (state: AppState) => ({
  analysisGame: state.analysisGame,
});

const mapDispatchToProps = {
  loadAnalysisGameById: (id: string) => loadFinishedGameById(id),
  loadAnalysisGameByDate: (date: Date) => loadFinishedGameByDate(date),
  loadLatestAnalysisGame: () => loadLatestFinishedGame(),
  moveForwardInHistory: (numberOfMoves: number) =>
    moveForwardInHistory(numberOfMoves),
  moveBackwardInHistory: (numberOfMoves: number) =>
    moveBackwardInHistory(numberOfMoves),
};

export default connect(mapStateToProps, mapDispatchToProps)(AnalysisGameRoute);
