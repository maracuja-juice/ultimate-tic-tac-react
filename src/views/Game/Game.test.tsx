import * as React from 'react';
import { renderWithStore, screen } from '../../test-utils';
import userEvent from '@testing-library/user-event';
import Game from './Game';

test('allows playing according to rules', () => {
  renderWithStore(<Game />);

  const buttons = screen.getAllByRole('button', { name: /^$/ });
  userEvent.click(buttons[0]);
  expect(screen.getByRole('button', { name: /x/i })).toBeInTheDocument();

  const buttonNotClickable = buttons[Math.floor(buttons.length / 2)];
  userEvent.click(buttonNotClickable);
  expect(screen.queryByRole('button', { name: /o/i })).not.toBeInTheDocument();

  userEvent.click(buttons[1]);
  expect(screen.getByRole('button', { name: /o/i })).toBeInTheDocument();
});
