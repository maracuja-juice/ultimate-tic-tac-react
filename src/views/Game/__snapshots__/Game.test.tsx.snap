// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should match snapshot 1`] = `
<div
  className="centerAll"
>
  <div
    className="gameWrapper"
  >
    <ConnectFunction />
    <BigBoard
      activeBoards={Array []}
      board={
        Array [
          Object {
            "position": Object {
              "x": 0,
              "y": 0,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 0,
          },
          Object {
            "position": Object {
              "x": 0,
              "y": 1,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 0,
          },
          Object {
            "position": Object {
              "x": 0,
              "y": 2,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 0,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 2,
          },
          Object {
            "position": Object {
              "x": 1,
              "y": 0,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 2,
          },
          Object {
            "position": Object {
              "x": 1,
              "y": 1,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 1,
          },
          Object {
            "position": Object {
              "x": 1,
              "y": 2,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 1,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 2,
          },
          Object {
            "position": Object {
              "x": 2,
              "y": 0,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 0,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 2,
          },
          Object {
            "position": Object {
              "x": 2,
              "y": 1,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 1,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 1,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 1,
          },
          Object {
            "position": Object {
              "x": 2,
              "y": 2,
            },
            "tiles": Array [
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 0,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 1,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 1,
                  "y": 2,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 0,
                },
                "value": 2,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 1,
                },
                "value": 0,
              },
              Object {
                "boardPosition": Object {
                  "x": 2,
                  "y": 2,
                },
                "position": Object {
                  "x": 2,
                  "y": 2,
                },
                "value": 2,
              },
            ],
            "value": 2,
          },
        ]
      }
      currentPlayer={0}
      onPlayerMoved={[MockFunction]}
    />
  </div>
</div>
`;
